tensorflow/compiler/mlir/xla/ir/hlo_ops.td

```diff
@@ -648,6 +648,27 @@ def HLO_CompareOp: HLO_Op<"compare",
   >];
 }
 
+def HLO_UniqueCountOp: HLO_UnaryElementwiseOp<"unique_count",
```

Comment:
This is effectively adding a new op and so should be documented here (e.g., what types, what input sizes)

How is unique_count element wise? Wouldn't it need to be across the entire input?

Does this op's operand and result type have to match? If so use that trait and then you don't need the custom builder.